packages/destination-actions/src/destinations/engage-messaging-sendgrid/utils/api-lookups.ts

```
@@ -0,0 +1,154 @@
+import { InputField } from '@segment/actions-core'
+import { RequestClient, RequestOptions } from '@segment/actions-core'
+import { Logger, StatsClient } from '@segment/actions-core/src/destination-kit'
+import type { Settings } from '../generated-types'
+import { Liquid as LiquidJs } from 'liquidjs'
+import { Profile } from './types'
+
+const Liquid = new LiquidJs()
+
+type ApiLookupConfig = {
+  id?: string | undefined
+  name: string
+  url: string
+  method: string
+  /** Cache ttl in ms */
+  cacheTtl: number
+  body?: string | undefined
+  headers?: object | undefined
+  responseType: string
+}
+
+export const performApiLookup = async (
+  request: RequestClient,
+  { id, url, method, body, headers }: ApiLookupConfig,
+  profile: Profile,
+  statsClient: StatsClient | undefined,
+  tags: string[],
+  settings: Settings,
+  logger?: Logger | undefined
+) => {
+  let renderedUrl: string
+  let renderedBody: string | undefined
+  try {
+    renderedUrl = await Liquid.parseAndRender(url, { profile })
+  } catch (error) {
+    logger?.error(
+      `TE Messaging: Email api lookup url parse failure - api lookup id: ${id} - ${settings.spaceId} - [${error}]`
+    )
+    tags.push('reason:parse_apilookup_url')
+    statsClient?.incr('actions-personas-messaging-sendgrid-error', 1, tags)
+    throw new IntegrationError('Unable to parse email api lookup url', 'api lookup url parse failure', 400)
+  }
+  try {
+    renderedBody = body ? await Liquid.parseAndRender(body, { profile }) : undefined
+  } catch (error) {
+    logger?.error(
+      `TE Messaging: Email api lookup body parse failure - api lookup id: ${id} - ${settings.spaceId} - [${error}]`
+    )
+    tags.push('reason:parse_apilookup_body')
+    statsClient?.incr('actions-personas-messaging-sendgrid-error', 1, tags)
+    throw new IntegrationError('Unable to parse email api lookup body', 'api lookup body parse failure', 400)
+  }
+
+  try {
+    const res = await request(renderedUrl, {
```

https://github.com/segmentio/action-destinations/blob/2abde6b72f8a5b52175389968ce1b8e15e79af3b/packages/destination-actions/src/destinations/engage-messaging-sendgrid/sendEmail/index.ts#L231

it might be worth adding `skipResponseCloning` to this. We had a sev in the past because certain responses would be too large and cause the requests to silently fail. I would double check to see if the bug that requires this work around was fixed but since we can't except certain response sizes it might be worth adding.